infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/PodMerger.java

```diff
@@ -141,4 +154,33 @@ private Deployment createEmptyDeployment(String name) {
         .endSpec()
         .build();
   }
+
+  private PodSecurityContext mergeSecurityContexts(
+      @Nullable PodSecurityContext a, @Nullable PodSecurityContext b) throws ValidationException {
+    return nonNullOrEqual(a, b, "Cannot merge pods with different security contexts: %s, %s");
+  }
+
+  private String mergeServiceAccount(@Nullable String a, @Nullable String b)
+      throws ValidationException {
+    return nonNullOrEqual(a, b, "Cannot merge pods with different service accounts: %s, %s");
+  }
+
+  private String mergeServiceAccountName(@Nullable String a, @Nullable String b)
+      throws ValidationException {
+    return nonNullOrEqual(a, b, "Cannot merge pods with different service account names: %s, %s");
+  }
+
+  @Nullable
+  private static <T> T nonNullOrEqual(@Nullable T a, @Nullable T b, String errorMessageTemplate)
```

Comment:
any reason it is static?

Reply:
It is private and doesn't use any instance variables.